csharp/src/Ice/Util.cs

```diff
@@ -160,32 +133,23 @@ public static Communicator initialize(ref string[] args, InitializationData init
     /// <returns>The initialized communicator.</returns>
     public static Communicator initialize(ref string[] args, string configFile)
     {
-        InitializationData initData = null;
-        if (configFile != null)
-        {
-            initData = new InitializationData();
-            initData.properties = new Properties();
-            initData.properties.load(configFile);
-        }
+        var initData = new InitializationData();
+        initData.properties = new Properties();
+        initData.properties.load(configFile);
         return initialize(ref args, initData);
     }
 
     /// <summary>
     /// Creates a communicator.
     /// </summary>
-    /// <param name="initData">Additional intialization data.</param>
+    /// <param name="initData">Additional initialization data.</param>
     /// <returns>The initialized communicator.</returns>
-    public static Communicator initialize(InitializationData initData)
+    public static Communicator initialize(InitializationData? initData = null)
     {
-        if (initData == null)
-        {
-            initData = new InitializationData();
-        }
-        else
-        {
-            initData = initData.Clone();
-        }
 
+        initData = initData is null ? new InitializationData() : initData with { };
+        // TODO: some tests rely on updating the properties after initialize.
```

Comment:
It's unfortunate that the "ref args" initialize and the without "ref args" initialize have different undocumented behaviors:

- with the "ref args" initialize, we copy all the properties and insert the Ice.ProgramName property ... and more related to Ice.Config
- with the no "ref args" initialize, we reference directly the initData.Properties (no copying), don't inset Ice.ProgramName and don't check Ice.Config

I don't see how to rationalize this behavior without breaking existing code.